### PR TITLE
Feature cc39 resolve login conflict

### PR DESCRIPTION
**Pull request by zhuocun**
image

Context: a user have registered an account with email, and then tries to login with google with the same email.
Feature: a new window will show and indicate user to connect account with google, after that, user can login with both way.

### CODE REVIEW

File: src/auth/auth.controller.ts [hunk 1]
```diff
@@ -2,8 +2,6 @@ import { Body, Controller, HttpCode, HttpStatus, Post, UseGuards } from "@nestjs
 import { ObjectId } from "mongoose";
 import { GetCurrentAdmin, GetCurrentAdminId } from "src/common/decorators";
 import { AccessTokenGuard, RefreshTokenGuard } from "src/common/guards";
-import { CreateUserDto } from "src/users/dto/createUser.dto";
```
Comment (zhuocun) [inline src/auth/auth.controller.ts:5]: These two are unused imports but don't have relationship with my card

File: src/users/user.service.ts [hunk 2]
```diff
@@ -86,10 +106,10 @@ export class UserService {
 
   /**
    * Check user existence by email
-   * @param {CheckEmailDto}
+   * @param emailDto
    */
   async checkEmail(emailDto: CheckEmailDto): Promise<boolean> {
     const foundUser = await this.userModel.find({ email: emailDto.email }).exec();
-    return foundUser.length > 0 ? true : false;
```
Comment (zhuocun) [inline src/users/user.service.ts:93]: this one is just to resolve IDE's syntax warning

File: src/users/user.service.ts [hunk 3]
```diff
@@ -57,13 +59,31 @@ export class UserService {
     }
     // Add new update date
     updateUserDto = { ...updateUserDto, updatedAt: new Date() };
-    const updatedUser = await await this.userModel
```
Comment (zhuocun) [inline src/users/user.service.ts:60]: this one is just to resolve IDE's warning

File: src/users/user.service.ts [hunk 4]
```diff
@@ -1,15 +1,17 @@
-import { HttpStatus, Injectable, NotFoundException } from "@nestjs/common";
```
Comment (zhuocun) [inline src/users/user.service.ts:1]: ununsed import

File: src/auth/auth.service.ts [hunk 5]
```diff
@@ -8,7 +8,6 @@ import * as argon from "argon2";
 import { JwtService } from "@nestjs/jwt";
 import { CreateUserDto } from "src/users/dto/createUser.dto";
 import { sendEmail } from "./emailHelpers";
-import { SendEmailCommand } from "@aws-sdk/client-ses";
```
Comment (zhuocun) [inline src/auth/auth.service.ts:11]: unused import

File: src/auth/auth.controller.ts [hunk 6]
```diff
@@ -2,8 +2,6 @@ import { Body, Controller, HttpCode, HttpStatus, Post, UseGuards } from "@nestjs
...
-import { User } from "src/users/schemas/user.schema";
```
Comment (zhuocun) [inline src/auth/auth.controller.ts:6]: these two are unused imports and warned by IDE

File: src/users/user.testData.ts [hunk 7]
```diff
@@ -68,3 +68,15 @@ export const userArray = [
     otpExpiresAt: null,
   },
 ];
+
+export const unconnectedAccount = {
```
Comment (Chanran) [inline src/users/user.testData.ts:72]: Maybe you should add more test cases such as googleId  is empty string, or googleId is numerical.

Comment (zhuocun) [reply src/users/user.testData.ts:72]: > Maybe you should add more test cases such as googleId is empty string, or googleId is numerical.

Thanks for reminding me of that! However, after going through other tests, I haven't found any of them checked the type of parameters. Maybe that's because Nest will help us do that based on decorators in "dto" files? Based on this thought, I updated the decorators of the ConnectAccountDto and hoped this could be another way to ensure the integrity of incoming parameters of requests.

**Review by jennawan — APPROVED** [review]
LGTM!
